Mathlib/RingTheory/LocalProperties.lean

```diff
@@ -3,14 +3,9 @@ Copyright (c) 2021 Andrew Yang. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Andrew Yang
 -/
-import Mathlib.RingTheory.FiniteType
-import Mathlib.RingTheory.Localization.AtPrime
-import Mathlib.RingTheory.Localization.Away.Basic
-import Mathlib.RingTheory.Localization.Integer
 import Mathlib.RingTheory.Localization.Submodule
-import Mathlib.RingTheory.Nilpotent.Lemmas
 import Mathlib.RingTheory.RingHomProperties
-import Mathlib.Data.Set.Subsingleton
+import Mathlib.RingTheory.IntegralClosure.IntegrallyClosed
```

Comment:
This seems very dubious.

Reply:
We can maybe split some file, but `RingTheory.LocalProperties` looks like a reasonable "advanced" file to import integrally closed ring. Did you notice a serious slowdown?

Reply:
No. I don't think so. But importing a specific local property into this file about local properties in general is a bad precedent. (Though not sure how much that has happened already.)

Reply:
Ah, I see. I didn't think to that.

Reply:
Sorry, I added the theorem `isIntegrallyClosed_ofLocalizationMaximal` in this file because I noticed that it already contains local properties such as `IsReduced`. It seems that letting the file `IntegrallyClosed.lean` import the file `LocalProperties.lean` is not an good choice either. So where would be the best place to put the theorem `isIntegrallyClosed_ofLocalizationMaximal`? Maybe we should split the section on local properties in general and the properties that satisfy local properties in `LocalProperties.lean` into separate files.

Reply:
There is no need to apologize! That indeed is very sound logic. I think splitting out specific local properties out into individual files is a good idea. I would be happy to merge such a PR. 

Thanks for the contribution. This is a very important fact we were missing.

Reply:
> There is no need to apologize! That indeed is very sound logic. I think splitting out specific local properties out into individual files is a good idea. I would be happy to merge such a PR.
> 
> Thanks for the contribution. This is a very important fact we were missing.

Thanks! I will split that file later.

Reply:
See PR #16879